jsettlers.logic/src/main/java/jsettlers/logic/map/loading/list/MapList.java

```diff
@@ -259,9 +259,14 @@ public MapList getMapList() {
 			return new MapList(getMapListers(), saveDirectory);
 		}
 
-		public void addResourcesDirectory(File resources) {
-			addMapDirectory(new DirectoryMapLister(new File(resources, "maps"), true));
-			saveDirectory = new DirectoryMapLister(new File(resources, "save"), true);
+		public void addResourcesDirectory(File savesFolder, File ... mapsFolders) {
+			for (File mapsFolder : mapsFolders) {
+				if (mapsFolder != null) {
```

Comment:
Don't use null array entries to indicate a missing entry. Just don't add it to the array then